chain/ethereum/src/ethereum_adapter.rs

```diff
@@ -1664,6 +1664,11 @@ async fn filter_call_triggers_from_unsuccessful_transactions(
             "failed to obtain transaction hash from call triggers"
         ))?;
 
+    // Return early if there are no transaction hashes
+    if transaction_hashes.is_empty() {
```

Comment:
What about moving the code that's below to here as well?

Like this:

```rust
if transaction_hashes.is_empty() || transactions.is_empty() {
    return Ok(block);
}
```

Reply:
Actually my idea will change behavior here: https://github.com/graphprotocol/graph-node/pull/3254/files#diff-b31a71a95f48aa7d0f11b5ea3501befc2560d749d73872e9eebe8003de85d921R1689

Reply:
It's not possible to move that check earlier because we still have to create the `transactions` value.

But, since we already checked that `transaction_hashes`:
1. is not empty (as this PR), and
2. has a different length than `transactions`,

it is safe to remove that check entirely, I believe.